tests/unit/test_url.py

```diff
@@ -0,0 +1,22 @@
+from pip._internal.index import _clean_link
```

Comment:
Since you're testing a function in `index.py`, you should put your tests in `test_index.py` instead of creating a new test module.